client/data/hooks/index.js

```diff
@@ -132,10 +101,11 @@ export const useAccountInfo = () => {
 	const getAccountInfo = async () => {
 		const info = await requestAccountInfo();
 		setAccountInfo( info );
+		return info;
```

Comment:
Technically not really necessary since when `accountInfo` is updated that automatically propagates down to wherever this hook is used. So you shouldn't need to rely on `reloadAccountInfo()` to return a value. The function name doesn't suggest it either.

Reply:
Completely agree, but did you see how it's used? Providing both data and a reloading mechanism makes for this scenario where you'd need to "try to connect again using the reloader, if successful, use the newly returned data". Not sure it can work as you suggest (which would be the optimal/react way of doing things)

Reply:
I see now, yeah... It probably makes a lot more sense for the entire hook to be replaced by a state branch with an appropriate _control_ and _resolver_ - in `@wordpress/data` terms.

But until we implement that, this should be ok.